src/harp_pedal_wizard.lua

```diff
@@ -0,0 +1,1591 @@
+                    if return_string.LuaString ~= "" then
+                        direct = true
+                        process_return(return_string.LuaString)
+                        strings_read()
+                    else
+                        if scale_check:GetCheck() == 1 then 
+                            scaleinfo = root.LuaString.." "..scales[sel_scale:GetSelectedItem() + 1]
+                        elseif chord_check:GetCheck() == 1 then 
+                            scaleinfo = root.LuaString..chords[sel_chord:GetSelectedItem() + 1]
+                        end
+                    end
+
+                    harp_diagram(str.LuaString, diag, scaleinfo, partial)
+                    if (changes == false) and (partial == true) then
+                        if override == false then
+--                            require('mobdebug').start()
+                            ui:AlertInfo("There are no pedal changes required. Try entering notes directly in the 'Enter Notes' field, or update your 'last used' pedals somewhere else and try again.", nil)
+                        end
+                        goto error
+                    end
+                    pedals_add(diag, partial)
+                    ::error::
+                    str.LuaString = ""
+                    harp_notes:SetText(str)
+                    changes_static:SetText(str)
+                    for i = 1, 6, 1 do
+                        str.LuaString = str.LuaString..harpstrings[i]..", "
+                    end
+                    str.LuaString = str.LuaString..harpstrings[7]
+                    config.last_notes = str.LuaString
+                    harp_config_save(config)
+                    finenv.Region():Redraw()
+                    direct = false
+                    override = false
+                    changes = false
+                    direct_notes = {0, 0, 0, 0, 0, 0, 0}
+                end -- apply()
+
+                dialog:RegisterHandleCommand(callback)
+                dialog:RegisterHandleOkButtonPressed(callback_ok)
+                dialog:RegisterHandleDataListSelect(callback_update)
+                if dialog.RegisterCloseWindow then
+                    dialog:RegisterCloseWindow(on_close)
+                end
+                -- initialize pedals
+                partial = false 
+                harp_diagram(config.last_notes, diag, scaleinfo, partial)
+                update_variables()
+
+                if partial_checkbox:GetCheck() == 1 then partial = true end
+                pedals_update()
+
+                dialog.OkButtonCanClose = false
+                finenv.RegisterModelessDialog(dialog) -- must register, or ShowModeless does nothing
+                dialog:ShowModeless()
+
+--                if dialog:ExecuteModal(nil) == finale.EXECMODAL_OK then
+--                end -- if dialog:Execute...
+
+            end -- get_harp
+            harp_dialog()
+        end -- harp
+
+        harp()
```

Comment:
I'm being a bit picking on this review because I really like this script. It's a great idea for a script, and I want to it to be as robust as possible. When RGP Lua 0.63 comes out I'm probably going to enhance it to use the `utf8` library for SMuFL handling (unless you want to). Also, once we can deploy mixins correctly, this definitely needs to transition to `RunModeless`.